### ISSUE TITLE
JDBC INSERT...RETURNING doesn't work on tables with UINTEGER columns

### ISSUE THREAD

**Issue by MariusVolkhart** [duckdb/duckdb#6301]
### What happens?

Given a table with a  `UINTEGER` column, the `RETURNING` clause of an `INSERT` statement returns the wrong numbers of columns, and the wrong value for columns.

A single column is returned with the value `1`. So, if expecting multiple columns, an `ArrayIndexOutOfBoundsException` is raised:
```java
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
	at org.duckdb.DuckDBResultSet.check_and_null(DuckDBResultSet.java:214)
	at org.duckdb.DuckDBResultSet.getString(DuckDBResultSet.java:235)
```

If expecting just a single column, the value is wrong.

### To Reproduce

Will link a branch with Java tests reproducing the problem. This does not happen using the DuckDB CLI. Can attach a patch or, I guess, copy/paste them here if preferred.

### OS:

MacOS x64

### DuckDB Version:

0.7.0

### DuckDB Client:

JDBC

### Full Name:

Marius Volkhart

### Affiliation:

PKWARE

### Have you tried this on the latest `master` branch?

- [X] I agree

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] I agree

**Comment by MariusVolkhart** [comment]
https://github.com/duckdb/duckdb/compare/master...MariusVolkhart:duckdb:mv/jdbcReturningUinteger?expand=1 contains the diff of tests that reproduce the issue and some minor code/CI changes that might help find issues like this in the future.

**Comment by MariusVolkhart** [comment]
Fixed by #6327
